test/napi/lib/threads.js

```diff
@@ -71,4 +71,17 @@ const assert = require('chai').assert;
     // Asynchronously GC to give the task queue a chance to execute
     setTimeout(() => global.gc(), 10);
   });
+
+  it('should be able to join on the result of a channel', function (cb) {
+    let msg = "Uninitialized";
```

Comment:
This and the Rust logic that goes with it took me a few minutes to follow 😂 

So `msg` is a shared mutable cell, we're modifying it after a 10ms timeout, and we're creating a Rust thread that waits 100ms and sends one event back to the main thread with the updated message, joins on that message, and sends another event back to the main thread that sends the message to an async callback. And then that callback tests that the message was the updated message.

Did I get that right? A comment might help 😅

Reply:
Yes, that's right! I'll write a more detailed comment that also explains _what_ we're trying to prove here. Specifically, we want to confirm that we can join on the result of `Channel::send` without impacting the JavaScript main thread.

Reply:
Added a bunch of comments. 😄